WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java

```diff
@@ -1743,6 +1748,150 @@ public boolean addMedia(Uri mediaUri) {
         }
     }
 
+
+    private void dismissProgressDialog(ProgressDialog progressDialog) {
+        if (progressDialog != null && progressDialog.isShowing()) {
+            try {
+                progressDialog.dismiss();
+            } catch (IllegalArgumentException e) {
+                // dialog doesn't exist
+            }
+        }
+    }
+
+    private class VideoOptimizationProgressListener implements org.m4m.IProgressListener {
```

Comment:
What do you think about moving all the video optimization code to a separate class? My concern is that `EditPostActivity` is getting very large and unwieldy.